creation/web_base/glidein_startup.sh

```diff
@@ -152,1932 +74,467 @@ do_start_all() {
+        echo "GLIDEIN_INITIALIZED 0"
+        # ...but be optimist, and leave advertise_only for the actual error handling script
+        echo "GLIDEIN_ADVERTISE_ONLY 0"
+        echo "GLIDEIN_CONDOR_TOKEN ${GLIDEIN_CONDOR_TOKEN}"
+        echo "# --- User Parameters ---"
+    } >> "${glidein_config}"; then
+        early_glidein_failure "Failed in updating '${glidein_config}'"
+    fi
+}
+
+_main(){
+
+    ################################
+    parse_options "$@"
+
+    ################################
+    # Code block used to set the slots_layout
+    # make sure to have a valid slots_layout
+    if (echo "x${slots_layout}" | grep -i fixed) >/dev/null 2>&1 ; then
+        slots_layout="fixed"
+    else
+        slots_layout="partitionable"
     fi
-}
-
-# use this for early failures, when we cannot assume we can write to disk at all
-# too bad we end up with some repeated code, but difficult to do better
-early_glidein_failure() {
-  error_msg="$1"
-
-  warn "${error_msg}"
-
-  sleep "${sleep_time}"
-  # wait a bit in case of error, to reduce lost glideins
-
-  glidein_end_time="$(date +%s)"
-  result="    <metric name=\"failure\" ts=\"$(date --date=@"${glidein_end_time}" +%Y-%m-%dT%H:%M:%S%:z)\" uri=\"local\">WN_RESOURCE</metric>
-    <status>ERROR</status>
-    <detail>
-     ${error_msg}
-    </detail>"
 
-  final_result="$(construct_xml "${result}")"
-  final_result_simple="$(basexml2simplexml "${final_result}")"
-  # have no global section
-  final_result_long="$(simplexml2longxml "${final_result_simple}" "")"
+    ################################
+    parse_arguments
 
-  glidien_cleanup
-
-  print_tail 1 "${final_result_simple}" "${final_result_long}"
-
-  exit 1
-}
-
-
-# use this one once the most basic ops have been done
-glidein_exit() {
-  # Removed lines about $lock_file (lock file for whole machine) not present elsewhere
-
-  gwms_process_scripts "$GWMS_DIR" cleanup "${glidein_config}"
-
-  global_result=""
-  if [ -f otr_outlist.list ]; then
-      global_result=$(cat otr_outlist.list)
-      chmod u+w otr_outlist.list
-  fi
+    ################################
+    # Code block used to generate the glidein UUID
+    if command -v uuidgen >/dev/null 2>&1; then
+        glidein_uuid="$(uuidgen)"
+    else
+        glidein_uuid="$(od -x -w32 -N32 /dev/urandom | awk 'NR==1{OFS="-"; print $2$3,$4,$5,$6,$7$8$9}')"
+    fi
 
-  ge_last_script_name=$(extract_parent_fname "$1")
-  result=$(extract_parent_xml_detail "$1")
-  final_result=$(construct_xml "${result}")
+    ################################
+    print_header "$@"
 
-  # augment with node info
-  final_result_simple=$(basexml2simplexml "${final_result}")
+    ################################
+    spawn_multiple_glideins
 
-  # Create a richer version, too
-  final_result_long=$(simplexml2longxml "${final_result_simple}" "${global_result}")
+    ########################################
+    # Code block used to make sure nobody else can write my files
```

Comment:
Same as earlier:
`# Make sure nobody else can write my files, in the Grid world I cannot trust anybody`